MovieApp/DataLayer/Clients/BaseApiClient.swift

```diff
@@ -0,0 +1,72 @@
+import Foundation
+
+class BaseApiClient {
+
+    private let baseUrl: String
+    let apiKey = "e24dd8d2f3822e3917d10c6570d7f574"
+
+    init(baseUrl: String) {
+        self.baseUrl = baseUrl
+    }
+
+    func get<T: Decodable>(
+        path: String,
+        queryParameters: [String: String?]? = nil,
+        completionHandler: @escaping (Result<T, RequestError>) -> Void
+    ) {
+        var urlString = "\(baseUrl)\(path)"
+
+        if let parameters = queryParameters {
+            var parameterString = "?"
+            for (index, (key, value)) in parameters.enumerated() {
```

Comment:
I do not think this is a good way to build the URL. The parameters will not be URL encoded. Please search how to properly include the parameters into url.